TestPointGetForTemporaryTable timed out after 5m0s

## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

<!-- a step by step guide for reproducing the bug. -->

### 2. What did you expect to see? (Required)

### 3. What did you see instead (Required)

```
 [2024/07/17 12:55:00.445 +00:00] [INFO] [store_cache.go:1058] ["store health status changed"] [storeID=1] [isSlow=true] [healthDetail="{ ClientSideSlowScore: 100, TiKVSideSlowScore: 0 }"]
[2024/07/17 12:55:02.415 +00:00] [INFO] [region_request.go:968] ["throwing pseudo region error due to no replica available"] [req-ts=451207665072209921] [req-type=Get] [region="{ region id: 2, ver: 1, confVer: 1 }"] [replica-read-type=leader] [stale-read=false] [request-sender="{rpcError:<nil>, replicaSelector: replicaSelectorV2{replicaReadType: leader, attempts: 11, cacheRegionIsValid: false, replicaStatus: [peer: 3, store: 1, isEpochStale: false, attempts: 10, attempts_time: 133.8µs, replica-epoch: 0, store-epoch: 0, store-state: resolved, store-liveness-state: reachable]}}"] [total-round-stats="{total-backoff: 4m15.2s, total-backoff-times: 39}"] [current-round-stats="{time: 1m13s, backoff: 1m12.9s, timeout: 30s, req-max-exec-timeout: 30s, retry-times: 10}"]
[2024/07/17 12:55:30.329 +00:00] [INFO] [store_cache.go:1058] ["store health status changed"] [storeID=1] [isSlow=false] [healthDetail="{ ClientSideSlowScore: 1, TiKVSideSlowScore: 0 }"]
[2024/07/17 12:55:30.339 +00:00] [INFO] [store_cache.go:1058] ["store health status changed"] [storeID=1] [isSlow=true] [healthDetail="{ ClientSideSlowScore: 100, TiKVSideSlowScore: 0 }"]
panic: test timed out after 5m0s
running tests:
	TestPointGetForTemporaryTable (5m0s)
goroutine 11736 [running]:
testing.(*M).startAlarm.func1() 
```

https://tiprow.hawkingrei.com/view/gs/pingcapprow/pr-logs/pull/pingcap_tidb/54673/fast_test_tiprow/1813554170221826048#1:build-log.txt%3A2013

### 4. What is your TiDB version? (Required)

<!-- Paste the output of SELECT tidb_version() -->

--- a/pkg/executor/batch_point_get_test.go
+++ b/pkg/executor/batch_point_get_test.go
@@ -206,8 +206,6 @@ func TestPointGetForTemporaryTable(t *testing.T) {
 
 	tk := testkit.NewTestKit(t, store)
 	tk.MustExec("use test")
-
-
 	tk.MustExec("drop table if exists t")
 	tk.MustExec("create global temporary table t1 (id int primary key, val int) on commit delete rows")
 	tk.MustExec("begin")

--- a/pkg/executor/batch_point_get_test.go
+++ b/pkg/executor/batch_point_get_test.go
@@ -23,6 +23,7 @@ import (
 
 	"github.com/pingcap/failpoint"
 	"github.com/pingcap/tidb/pkg/executor"
+	"github.com/pingcap/tidb/pkg/infoschema"
 	"github.com/pingcap/tidb/pkg/kv"
 	"github.com/pingcap/tidb/pkg/sessionctx/variable"
 	"github.com/pingcap/tidb/pkg/testkit"
@@ -201,16 +202,23 @@ func TestCacheSnapShot(t *testing.T) {
 }
 
 func TestPointGetForTemporaryTable(t *testing.T) {
-	store := testkit.CreateMockStore(t)
+	store, dom := testkit.CreateMockStoreAndDomain(t)
 
 	tk := testkit.NewTestKit(t, store)
 	tk.MustExec("use test")
+
+
 	tk.MustExec("drop table if exists t")
 	tk.MustExec("create global temporary table t1 (id int primary key, val int) on commit delete rows")
 	tk.MustExec("begin")
 	tk.MustExec("insert into t1 values (1,1)")
 	tk.MustQuery("explain format = 'brief' select * from t1 where id in (1, 2, 3)").
 		Check(testkit.Rows("Batch_Point_Get 3.00 root table:t1 handle:[1 2 3], keep order:false, desc:false"))
+
+	isV2, _ := infoschema.IsV2(dom.InfoSchema())
+	if isV2 {
+		t.Skip("This test can not run under infoschema v2, because the later would always visit network")
+	}
 
 	require.NoError(t, failpoint.Enable("github.com/pingcap/tidb/pkg/store/mockstore/unistore/rpcServerBusy", "return(true)"))
 	defer func() {

